include/mq/api/MacroDataTypes.h

```diff
@@ -0,0 +1,553 @@
+	MQVARPTR_PROPERTIES(double, Double);
+	MQVARPTR_PROPERTIES(int64_t, Int64);
+	MQVARPTR_PROPERTIES(uint64_t, UInt64);
+	MQVARPTR_PROPERTIES(eqlib::ItemPtr, Item);
+#undef MQVARPTR_PROPERTIES
+
+	// TODO: Future work -- uncomment the deprecates and refactor all uses of high/low part
+	// these are here only to support deprecated functionality where some types rely on storing ints in low and high parts
+	//DEPRECATE("Use Get<uint32_t>() instead of LowPart. For data needing High and Low part, create a data structure instead.")
+	uint32_t get_LowPart() const { return Get<uint32_t>(); }
+
+	//DEPRECATE("Use Set<uint32_t>(v) instead of LowPart. For data needing High and Low part, create a data structure instead.")
+	uint32_t set_LowPart(uint32_t Val)
+	{
+		// we can assume that if the user wanted to also set HighPart, they did it explicitly either before or after setting
+		// LowPart, which falls into 2 cases. Either the underlying data was set before the 64-bit, or it wasn't (either the
+		// the default or an explicit 32-bit/pointer value).
+
+		// first check if we have 64-bit Data -- we only need to worry about unsigned because this trick wouldn't work with signed
+		// data without some explicit or implicit casting to unsigned
+		if (Data.index() == static_cast<size_t>(VariantIdx::UInt64))
+		{
+			// if so, we need to save the HighPart off before setting the low part, then retain the data type in the variant
+			ULARGE_INTEGER i;
+			i.QuadPart = Get<uint64_t>();
+			i.LowPart = Val;
+			Set<uint64_t>(i.QuadPart);
+			return i.LowPart;
+		}
+
+		// if we get here, we don't have 64-bit data, so HighPart is either unset, or set in the convenience variable above
+		// we can just set the LowPart as uint32_t
+		return Set<uint32_t>(Val);
+	}
+
+#pragma warning(suppress: 4996)
+	__declspec(property(get = get_LowPart, put = set_LowPart)) uint32_t LowPart;
+
+	//DEPRECATE("For data needing High and Low part, create a data structure instead.")
+	uint32_t get_HighPart() const
+	{
+		// this data is only stored in Data if the type of variant is UInt64
+		if (Data.index() == static_cast<size_t>(VariantIdx::UInt64))
+		{
+			ULARGE_INTEGER i;
+			i.QuadPart = Get<uint64_t>();
+			return i.HighPart;
+		}
+
+		// otherwise, just return the extra storage information
+		return HighPart_;
+	}
+
+	//DEPRECATE("For data needing High and Low part, create a data structure instead.")
+	uint32_t set_HighPart(uint32_t Val)
+	{
+		// again, check if we  have already set the underlying data type to uint64_t
+		if (Data.index() == static_cast<size_t>(VariantIdx::UInt64))
+		{
+			ULARGE_INTEGER i;
+			i.QuadPart = Get<uint64_t>();
+			i.HighPart = Val;
+			Set<uint64_t>(i.QuadPart);
+			return i.HighPart;
+		}
+
+		// otherwise, just set the extra storage information and move on
+		HighPart_ = Val;
+		return HighPart_;
+	}
+
+#pragma warning(disable: 4996)
+	__declspec(property(get = get_HighPart, put = set_HighPart)) uint32_t HighPart;
+#pragma warning(default: 4996)
```

Comment:
Why suppress above and disable/default here?  Leftover from the original copy?

Reply:
guess so. will fix up